### PR TITLE
Clean up usage of torch._six partially

### PR DESCRIPTION
**Pull request by skyline75489**
See #42919

### CODE REVIEW

**Comment by skyline75489** [comment]
Hi @bugra I was trying to make the CI happy I don't really understand what the `mypy failed` means. I'm not sure if it's caused by my change. By the look of it I didn't change the `__func__` part at all.

**Comment by bugra** [comment]
@skyline75489  Can you explain the issue that you are seeing? I did not follow.

**Comment by skyline75489** [comment]
@bugra So I was seeing a weird error:

```plaintext
 mypy failed: torch/jit/_script.py:630: error: "Callable[[], Any]" has no attribute "__func__" [attr-defined]
```

To avoid the error I had to restore `get_function_from_type` in  https://github.com/pytorch/pytorch/pull/49785/commits/d86e3f45850497eaad77ba62bc5da6ab57800a2b .

**Comment by bugra** [comment]
> @bugra So I was seeing a weird error:
> 
> ```
>  mypy failed: torch/jit/_script.py:630: error: "Callable[[], Any]" has no attribute "__func__" [attr-defined]
> ```
> 
> To avoid the error I had to restore `get_function_from_type` in [d86e3f4](https://github.com/pytorch/pytorch/commit/d86e3f45850497eaad77ba62bc5da6ab57800a2b) .

Oh, we can I think move this function to another util class rather than keeping it in here, I do not believe it has anything to do with compatibility of Python 2/3. @suo  What do you think?

**Comment by skyline75489** [comment]
@bugra I've moved `get_function_from_type` into `torch/jit/script.py` which is the only place that still uses it. Largely this PR is ready for review.

**Comment by bugra** [comment]
@skyline75489  Thanks for doing this! Can you rebase from master? Otherwise, it looks good.

**Comment by skyline75489** [comment]
@bugra Done. No big conflicts. Just a small one. Hope the CI is happy 😃

**Review by bugra — APPROVED** [review]

**Comment by bugra** [comment]
Looks good!

File: torch/jit/_script.py
Comment (albanD) [inline torch/jit/_script.py:57]: nit: not sure if this function is really necessary? Could you just have the two getattr calls below?

**Review by albanD — COMMENTED** [review]
looks good.
Just one question for the default value of a getattr and good to go otherwise :)

File: torch/jit/_script.py
Comment (skyline75489) [reply torch/jit/_script.py:57]: I did try that but the CI failed. So I restored it in https://github.com/pytorch/pytorch/commit/d86e3f45850497eaad77ba62bc5da6ab57800a2b

See https://github.com/pytorch/pytorch/pull/49785#issuecomment-752850244

File: torch/cuda/amp/grad_scaler.py
Comment (skyline75489) [reply torch/cuda/amp/grad_scaler.py:2]: I’m not really sure which one should be preferred: container.abc or container_abc. So I kinda mixed the usage to minimize my modification.

File: torch/jit/_recursive.py
Comment (skyline75489) [reply torch/jit/_recursive.py:498]: Yeah I’ll fix this later

**Comment by skyline75489** [comment]
Hi @bugra , anything stopping the landing of this PR?

**Comment by bugra** [comment]
@skyline75489 Internal usage is complicated, unfortunately I could not land as is. I need to fix some of the internal usage before we land this. I will update next week with a better ETA.

**Comment by skyline75489** [comment]
@bugra No worries. I fully understand. Take your time 😃

**Comment by bugra** [comment]
@skyline75489 This is now landed. Let's continue removing it from the codebase :)

**Pull request closed without merging**
